src/api/Metadata.js

```diff
@@ -926,7 +926,12 @@ class Metadata extends File {
                 const key = this.getMetadataCacheKey(id);
                 const cachedMetadata = cache.get(key);
 
-                const templateInstance = { ...template, type: metadata.data.$type };
+                const templateId = cachedMetadata.templates.find(
```

Comment:
What if no matching template is found?

Reply:
Good point. It shouldn't be a common case, but in that scenario I think we should use existing `templateInstance.id` as a fallback